src/generic_core/uproxy_core.ts

```diff
@@ -691,7 +691,7 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
     ui.update(uproxy_core_api.Update.CORE_UPDATE_AVAILABLE, details);
   }
 
-  public cloudUpdate = (args: uproxy_core_api.CloudOperationArgs): Promise<void> => {
+  public cloudUpdate = (args: uproxy_core_api.CloudOperationArgs): Promise<any> => {
```

Comment:
Neat :-) However, I shouldn't have recommended `any`. Let's give ourselves some type safety. Can you define a **return type** for this method? We can call it `CloudOperationResult` and for now it can have a single field, `hasOAuth`, which is set to true iff the `CloudOperationType` was `CLOUD_HAS_OAUTH` and we have an OAuth token.

Sorry, I should have thought of that - but you shouldn't have to update any of the other calls to `cloudUpdate`.